conans/client/graph/proxy.py

```diff
@@ -92,6 +92,18 @@ def _get_recipe(self, reference, remotes, update, check_update):
                 status = RECIPE_INCACHE_DATE_UPDATED
         return recipe_layout, status, remote
 
+    @staticmethod
+    def _should_update(reference, update):
+        if update is None:
+            return False
+        assert isinstance(update, list)
```

Comment:
I'd try to keep the type instead of list and do ``if update is True: return True``